tests/xharness/Hardware/Devices.cs

```diff
@@ -62,6 +62,31 @@ Task ILoadAsync.LoadAsync (ILog log, bool include_locked, bool force)
 			return LoadAsync (log, extra_data: false, removed_locked: !include_locked, force: force);
 		}
 
+		Device GetDevice (XmlNode deviceNone)
+		{
+			// get data, if we are missing some of them, we will return null, happens sometimes that we 
+			// have some empty nodes.
+			var usable = deviceNone.SelectSingleNode ("IsUsableForDebugging")?.InnerText;
+			try {
+				var device = new Device {
+					DeviceIdentifier = deviceNone.SelectSingleNode ("DeviceIdentifier")?.InnerText,
+					DeviceClass = (DeviceClass) Enum.Parse (typeof (DeviceClass), deviceNone.SelectSingleNode ("DeviceClass")?.InnerText, true),
+					CompanionIdentifier = deviceNone.SelectSingleNode ("CompanionIdentifier")?.InnerText,
+					Name = deviceNone.SelectSingleNode ("Name")?.InnerText,
+					BuildVersion = deviceNone.SelectSingleNode ("BuildVersion")?.InnerText,
+					ProductVersion = deviceNone.SelectSingleNode ("ProductVersion")?.InnerText,
+					ProductType = deviceNone.SelectSingleNode ("ProductType")?.InnerText,
+					InterfaceType = deviceNone.SelectSingleNode ("InterfaceType")?.InnerText,
+					IsUsableForDebugging = usable == null ? (bool?) null : ((bool?) (usable == "True")),
+				};
+				bool.TryParse (deviceNone.SelectSingleNode ("IsLocked")?.InnerText, out var locked);
+				device.IsLocked = locked;
+				return device;
+			} catch {
```

Comment:
A solution that doesn't involve throwing and catching exceptions is much better. One example is that exceptions tracing can be used to figure out real problems.

Reply:
Fixed, the TryParse should work correctly, if the DeviceClass cannot be parsed, return null.